### PR TITLE
Dynamos PIP package and re-design msChain communication

### PR DESCRIPTION
**Pull request by Jorrit05**
Goal:
Reduce complexity and streamline design

Did:
- Added a python package 'dynamos' to handle Microservice chain and Dynamos related functionality
- The Library implementation in Go and Python for handling Microservice chains are now almost equal
- re-designed the logic around sending AMQ messages to a microservice of a MS chain
- Some refactoring and renaming
- Added comments mostly in the sidecar and Python Dynamos Pip package

### CODE REVIEW

File: docs/installation_guide.md
Comment (aleandromifsud) [inline docs/installation_guide.md:1]: How come this file was added here too?

File: go/cmd/sql-anonymize/application_logic.go
Comment (aleandromifsud) [inline go/cmd/sql-anonymize/application_logic.go:1]: Very good idea to separate the app logic

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # If you prefer the allow list template instead of the deny list, see community template:
 # https://github.com/github/gitignore/blob/main/community/Golang/Go.AllowList.gitignore
 #
+build
+dist
+dynamos.egg.info/*
+*dynamos.egg-info
```
[the remark this answers is not in the document]
Comment (Jorrit05) [reply .gitignore:7]: If you build a pip package you get this .egg folder. It doesn't need to be checked into source control

File: charts/agents/values.yaml [hunk 2]
```diff
@@ -2,5 +2,5 @@ Namespace: uva
 etcdDns: etcd-headless.core.svc.cluster.local
 etcdEndpoints: "http://etcd-0.etcd-headless.core.svc.cluster.local:2379,http://etcd-1.etcd-headless.core.svc.cluster.local:2379,http://etcd-2.etcd-headless.core.svc.cluster.local:2379"
 tracingEndpoint: "collector.linkerd-jaeger:55678"
-branchNameTag: "main"
+branchNameTag: "pythonlib"
```
[the remark this answers is not in the document]
Comment (Jorrit05) [reply charts/agents/values.yaml]: Si si senor

File: docs/installation_guide.md
Comment (Jorrit05) [reply docs/installation_guide.md:1]: Merged main into my branch of course ;)

**Pull request merged**